File: NOVEMBER-2025/08-11-2025/1611.php
class Solution {
    public function minimumOneBitOperations($n) {
        $v = array_fill(0, 31, 0);
        $v[0] = 1;
        for ($i = 1; $i < 31; $i++) {
            $v[$i] = (2 * $v[$i - 1]) + 1;
        }

        $cnt = 0;
        $ans = 0;
        for ($i = 30; $i >= 0; $i--) {
            if ((1 << $i) & $n) {
                $cnt++;
                if ($cnt % 2) {
                    $ans += $v[$i];
                } else {
                    $ans -= $v[$i];
                }
            }
        }
        return $ans;
    }
}

File: NOVEMBER-2025/08-11-2025/1611.py
#python

class Solution(object):
    def minimumOneBitOperations(self, n):
        count = 0
        while n>0:
            count ^= n
            n >>= 1
        return count


#python3

class Solution:
    def minimumOneBitOperations(self, n: int) -> int:
        ans = 0
        while n:
            ans ^= n
            n >>= 1
        return ans